packages/flutter_driver/test/src/real_tests/flutter_driver_test.dart

```
@@ -24,10 +27,107 @@ const String _kWebScriptSuffix = "')";
 
 void main() {
   final List<String> log = <String>[];
+
   driverLog = (String source, String message) {
     log.add('$source: $message');
   };
 
+  group('VMServiceFlutterDriver with logCommunicationToFile', () {
+    late FakeVmService fakeClient;
+    FakeVM fakeVM;
+    late FakeIsolate fakeIsolate;
+    late VMServiceFlutterDriver driver;
+    int driverId = 0;
+    setUp(()  async {
+      _cleanTestOutputFiles();
+      fakeIsolate = FakeIsolate();
+      fakeVM = FakeVM(fakeIsolate);
+      fakeClient = FakeVmService(fakeVM);
+      fakeClient.responses['waitFor'] = makeFakeResponse(<String, dynamic>{'status':'ok'});
+    });
+
+    tearDown(() {
+      driverId++;
+    });
+
+    group('logCommunicationToFile', (){
+        test('logCommunicationToFile = true', () async {
+        // Given
+        driver = VMServiceFlutterDriver.connectedTo(fakeClient, fakeIsolate);
+
+        // When
+        await driver.waitFor(find.byTooltip('foo'), timeout: _kTestTimeout);
+
+        // Then
+        final File file = File(p.join(testOutputsDirectory, 'flutter_driver_commands_$driverId.log'));
+        // ignore: avoid_slow_async_io
```

don't ignore errors, instead fix them. (use existsSync)